src/url.js

```diff
@@ -649,3 +649,18 @@ export function checkCorsUrl(url) {
 export function tryDecodeUriComponent(component, opt_fallback) {
   return tryDecodeUriComponent_(component, opt_fallback);
 }
+
+/**
+ * Extract and return the domain from the given url string via regex
+ * Ex.  'http://google.com?test' -> 'http://google.com'
+ *      'https://google.com/test1/test2' -> 'https://google.com'
+ *      'http://localhost:8000/test' -> 'http://localhost:8000'
+ * @param {string} url
+ * @return {string|undefined}
+ */
+export function extractDomain(url) {
+  // regex adapted from https://stackoverflow.com/a/8498629
```

Comment:
Why are you using a regex instead of `url.origin`?

Reply:
Ended up using `url.origin` instead of this function